.github/workflows/test.yml

```
@@ -46,7 +46,19 @@ jobs:
           restore-keys: |
             ${{ env.cache-version }}-${{ runner.os }}--electron-builder-cache-
 
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        id: cache-node-modules
+        with:
+          path: node_modules
+          key: ${{ env.cache-version }}-${{ runner.os }}--node-modules-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ env.cache-version }}-${{ runner.os }}--node-modules-
+      - run: cp package-lock.json package-lock.json.bak
+      - run: npm i
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
       - run: npm ci
+        if: hashFiles('package-lock.json') != hashFiles('package-lock.json.bak')
```

```suggestion
      - run: npm ci
```

他のyarnを使っているリポジトリを見た感じ、以下のような構造が一般的なようでした。

https://github.com/tokyo-metropolitan-gov/covid19/blob/4681c98001a84da396d7123fb92072e94df25f25/.github/workflows/build.yml#L18-L32

`yarn install --frozen-lockfile`と同じことができれば、`npm i`と`npm ci`を重ねる必要はないはずです。
調べてみたところ、以下の情報が見つかりました。
https://stackoverflow.com/questions/63187000/what-is-the-npm-equivalent-of-yarn-install-frozen-lockfile

よって、`npm ci`をキャッシュヒット関係なく実行してしまえばなにも問題ないはずです。

下調べ不足で適当でないレビューをしてしまいました...

---

横槍すみません！

念のため前提知識を共有すると、
`npm ci`すると、node_modulesディレクトリを全部消して、１から全部再インストールするんですよね。。
なので`node_modules`をキャッシュすると良いのでは、というのが根本の課題です。

---

すみません、私の勘違いですね...
確かに`npm ci`は完全にフォルダを消してしまうので、時間がかかる問題は避けられないですね...
キャッシュヒットしなかった時だけciを実行する現在の状態が良いと思います...!